addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -617,6 +617,7 @@ publicWidget.registry.CarouselBootstrapUpgradeFix = publicWidget.Widget.extend({
         "[data-snippet='s_image_gallery'] .carousel",
         "[data-snippet='s_carousel'] .carousel",
         "[data-snippet='s_quotes_carousel'] .carousel",
+        "#o-carousel-product.carousel",
```

Comment:
Since this new case is not linked to a snippet, I think we should review what is below. For instance, there is: `const snippetName = this.el.closest("[data-snippet]").dataset.snippet;`. In the default case, we don't go through this line as the `o-carousel-product` has a data-ride. But if someone somehow removed the data-bs-ride and left a data-bs-duration he used during testing, the bootstrap behavior would be: never auto-cyling, which may have been what the user wanted and saw his issue fixed. But then you add this patch... and suddenly there is a crash: `this.el.closest("[data-snippet]").dataset` will crash.

In this case, I think just `this.el.closest("[data-snippet]").dataset` -> `this.el.closest("[data-snippet]")?.dataset`, would be fine :+1: 

To see if more needed.

Could you also add a `// TODO adapt the shop XML directly in master` next to you added line? :pray: 

Thanks :pray:

Reply:
Thanks for pointing this case out @qsm-odoo ^^ 
I've went with adding the `?.` for now.